distributionResources/manual-upgrade.sh

```diff
@@ -346,6 +346,23 @@ if [ $? != 0 ]; then print_error; fi # exit if the last command failed
 cp -R pipeline-lib $labkey_home
 if [ $? != 0 ]; then print_error; fi # exit if the last command failed
 
+#
+# Remove all existing LabKey JAR files from TOMCAT_HOME/lib directory, if found
+#
+jars=( "javax.activation.jar" "labkeyBootstrap.jar" "mysql.jar" "jtds.jar" "mail.jar" "postgresql.jar"); 
+for jar in "${jars[@]}"
+do
+    file=$CATALINA_HOME/lib/$jar
+    if [ -f $file ]
+    then
+        echo "Deleting the $file file..."
+        rm $file;
+        ret_value=$?
+        if [ $ret_value -ne 0 ]; then
+            echo "FAILED"
```

Comment:
I think failure to remove these jar files is possibly fatal, right @labkey-adam?  So perhaps we want to use print_error either here or after the for loop if any of the deletions has failed, which will cause the upgrade to fail.

Reply:
I don't know if I'd say truly fatal... they could upgrade and remove them manually afterwards. The server will be quite cranky at startup if it sees them, so admins will be well aware.

Reply:
They won't know that it's going to be cranky about this failure, though, so I think we should at least warn them that they'll need to do this manually if something did not get deleted.

Reply:
I agree we should warn

Reply:
I don't think a warning is necessary, not in that exact language anyway. The point of this part of the script is to remove the old JAR files in order for the next part to happen: Adding the new JAR files.

Previously, we never removed old JAR files and instead the script would just do a forced copy of all six JAR files from tomcat-lib to the CATALINA_HOME/lib directory. This was fine the old way when we needed to have all six of those JARs going to CATALINA_HOME/lib. But since we're planning on putting the database JARs under the labkeywebapp directory, we need something that will always remove those old DB JAR files out of the CATALINA_HOME/lib during the upgrade process so it won't cause a conflict with the new DB JAR file location under labkeywebapp.

This is why my version here only indicates that a file was skipped because it doesn't exist (https://github.com/LabKey/gradlePlugin/pull/54#issuecomment-600381245), which is okay for the newer versions of LabKey that will be putting those JARs under labkeywebapp and if anyone uses this newer upgrade script on older binaries, it still will drop in the six JAR files to CATALINA_HOME/lib since the forced copy line is still in the script after it does the delete.

Reply:
I would prefer if we add a warning if one of the legacy files that is meant to be deleted and not replaced cannot be deleted for some reason.  It's unlikely to happen, but could and it just seems better to provide as much information as we can.

Just a small modification to what is here:
```
jars=( "javax.activation.jar" "labkeyBootstrap.jar" "mysql.jar" "jtds.jar" "mail.jar" "postgresql.jar"); 

deleteFailed=0
for jar in "${jars[@]}"
do
    file=$CATALINA_HOME/lib/$jar
    if [ -f $file ]
    then
        echo "Deleting the $file file..."
        rm $file;
        retValue=$?
        if [ $retValue -ne 0 ]; then
            echo "FAILED"
            deleteFailed=1
        fi
    fi
done

if [ $deleteFailed -eq 1 ]; then
    echo "Unable to delete one or more legacy jar files from $CATALINA_HOME/lib.  You will need to remove these manually in order for the server to start cleanly."
fi
```

Reply:
That looks good to me. It is unlikely, but this definitely covers us if it ever does occur.

Reply:
I did a quick test with the revisions that Susan did.

If the files exist, it'll state that the files are being deleted.

If the files don't exist, it won't output anything, which is fine.

I tried to change the attributes on one of my JAR files to make it immutable and then ran the script. However, it didn't output a delete failed, but it gave me a prompt asking if I wanted to delete the file I had set to being immutable.

Reply:
Add a `-f` to the `rm` command to get rid of that prompt (`rm -f $file`)